internal/services/containers/kubernetes_cluster_data_source_test.go

```diff
@@ -554,26 +554,6 @@ func TestAccDataSourceKubernetesCluster_microsoftDefender(t *testing.T) {
 	})
 }
 
-func TestAccDataSourceKubernetesCluster_customCaTrustCerts(t *testing.T) {
```

Comment:
We shouldn't be removing these tests until after 4.0
- We still need the ability to run these in case there are severe issues with any features that are still available in 3.x
- In case of unforeseen consequences that may require us to roll back anything

Can you please just skip the tests using the flag for now. We can worry about cleaning up unused tests and configs post major release.

Reply:
Got it! I've updated this PR as suggested.